tests/server.py

```diff
@@ -593,7 +593,7 @@ def flush(self, maxbytes=None):
         if self.disconnected:
             return
 
-        if getattr(self.other, "transport") is None:
+        if hasattr(self.other, "transport"):
```

Comment:
From [this review chain](https://github.com/matrix-org/synapse/pull/9649#discussion_r597986898) I assume `hasattr` was originally intended here, instead of `getattr`.